src/mds/Server.cc

```diff
@@ -3750,6 +3750,8 @@ void Server::handle_client_setattr(MDRequestRef& mdr)
     if (truncating_smaller && pi->is_truncating()) {
       dout(10) << " waiting for pending truncate from " << pi->truncate_from
 	       << " to " << pi->truncate_size << " to complete on " << *cur << dendl;
+      mds->locker->drop_locks(mdr.get());
```

Comment:
This doesn't seem unreasonable, but why do it here and not in the other place we use WAIT_TRUNC above?